source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1237 @@
+          prm.declare_entry ("Use marine component", "false",
+                             Patterns::Bool (),
+                             "Flag to use the marine component of FastScape.");
+          prm.declare_entry ("Use stratigraphy", "false",
+                             Patterns::Bool (),
+                             "Flag to use stratigraphy");
+          prm.declare_entry("Total steps", "100000",
+                            Patterns::Integer(),
+                            "Total number of steps you expect in the FastScape model, only used if stratigraphy is turned on.");
+          prm.declare_entry("Number of horizons", "1",
+                            Patterns::Integer(),
+                            "Number of horizons to track and visualize in FastScape.");
+          prm.declare_entry("Y extent in 2d", "100000",
+                            Patterns::Double(),
+                            "Y extent when used with 2D");
+          prm.declare_entry ("Use ghost nodes", "true",
+                             Patterns::Bool (),
+                             "Flag to use ghost nodes");
+          prm.declare_entry ("Use velocities", "true",
+                             Patterns::Bool (),
+                             "Flag to use FastScape advection and uplift.");
+          prm.declare_entry("Precision", "0.001",
+                            Patterns::Double(),
+                            "How close an ASPECT node needs to be to a FastScape node.");
+          prm.declare_entry ("Sediment rain", "0,0",
+                             Patterns::List (Patterns::Double(0)),
+                             "Sediment rain values given as a list equal to the number of intervals.");
+          prm.declare_entry ("Sediment rain intervals", "0",
+                             Patterns::List (Patterns::Double(0)),
+                             "A list of sediment rain times.");
+
+
+          prm.enter_subsection ("Boundary conditions");
+          {
+            prm.declare_entry ("Bottom", "1",
+                               Patterns::Integer (0, 1),
+                               "Bottom boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Right", "1",
+                               Patterns::Integer (0, 1),
+                               "Right boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Top", "1",
+                               Patterns::Integer (0, 1),
+                               "Top boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Left", "1",
+                               Patterns::Integer (0, 1),
+                               "Left boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry("Left mass flux", "0",
+                              Patterns::Double(),
+                              "Flux per unit length through left boundary (m^2/yr)");
```

Comment:
Please use the correct formatting of units here and in all `declare_entry`'s, e.g. `Units: $\\si{m^2/s}$.`.